### PR TITLE
[webview_flutter] In the case of sw backend, use only one buffer for rendering

### PR DESCRIPTION
**Pull request by bwikbs**
This is a compromise of partial rendering for performance. A sync issue may occur in the sw backend.

This patch improves rendering which is related to issue 2 of https://github.com/flutter-tizen/plugins/issues/339

### CODE REVIEW

File: packages/webview_flutter/tizen/src/webview.cc [hunk 1]
```diff
@@ -164,14 +165,20 @@ WebView::WebView(flutter::PluginRegistrar* registrar, int viewId,
...
-      rendered_surface_(nullptr),
       is_mouse_lbutton_down_(false),
+      rendered_surface_(nullptr),
       has_navigation_delegate_(false),
       has_progress_tracking_(false),
       context_(nullptr),
       texture_variant_(nullptr),
       platform_window_(platform_window) {
-  tbm_pool_ = std::make_unique<BufferPool>(width, height);
+  use_sw_backend_ = needsSwBackend();
```
Comment (seungsoo47) [inline packages/webview_flutter/tizen/src/webview.cc:175]:
```suggestion
  use_sw_backend_ = NeedsSwBackend();
```
Please change the 'n' to uppercase.

File: packages/webview_flutter/tizen/src/webview.cc [hunk 2]
```diff
@@ -19,6 +19,7 @@
 #include "lwe/PlatformIntegrationData.h"
 #include "webview_factory.h"
 
+#define BUFFER_POOL_SIZE 5
```
Comment (seungsoo47) [inline packages/webview_flutter/tizen/src/webview.cc:22]: Why did you declare it to 5?

Comment (bwikbs) [reply packages/webview_flutter/tizen/src/webview.cc:22]: It is not the value set this time, it is the value it already had.
There was no particular reason.
It was moderately selected for hands greater than 3. 
at multiful webviews, if frequent updates such as animations occur, buffer shortages may occur, so a moderately large value was chosen.

File: packages/webview_flutter/tizen/src/webview.cc (hunk 1 above)
Comment (bwikbs) [reply packages/webview_flutter/tizen/src/webview.cc]: Good point! done!

File: packages/webview_flutter/tizen/src/webview.cc [hunk 3]
```diff
@@ -164,14 +165,20 @@ WebView::WebView(flutter::PluginRegistrar* registrar, int viewId,
...
       context_(nullptr),
       texture_variant_(nullptr),
       platform_window_(platform_window) {
-  tbm_pool_ = std::make_unique<BufferPool>(width, height);
+  use_sw_backend_ = NeedsSwBackend();
+  if (use_sw_backend_) {
+    tbm_pool_ = new SingleBufferPool(width, height);
+  } else {
+    tbm_pool_ = new BufferPool(width, height, BUFFER_POOL_SIZE);
+  }
```
Comment (bbrto21) [inline packages/webview_flutter/tizen/src/webview.cc:180]: Why don't you use [unique_ptr](https://en.cppreference.com/w/cpp/memory/unique_ptr) anymore?

``` c++
std::cout << "\n" "2) Runtime polymorphism demo\n";
{
    // Create a derived resource and point to it via base type
    std::unique_ptr<B> p = std::make_unique<D>();

    // Dynamic dispatch works as expected
    p->bar();
}
```

Comment (bwikbs) [reply packages/webview_flutter/tizen/src/webview.cc:180]: Well, there's no other reason. When I first did it, an error occurred and I just thought it wouldn't work.
I'll try it again.

Comment (bwikbs) [reply packages/webview_flutter/tizen/src/webview.cc]: done!

**Review by bbrto21 — APPROVED** [review]

File: packages/webview_flutter/tizen/src/webview.cc (hunk 2 above)
Comment (seungsoo47) [reply packages/webview_flutter/tizen/src/webview.cc:22]: I understood. Thanks for the details. 👍

**Review by seungsoo47 — APPROVED** [review]

**Pull request merged**